deployer/grafana/grafana_utils.py

```diff
@@ -0,0 +1,149 @@
+    Returns:
+        string object: https address of the prometheus instance
+    Raises ValueError if:
+        - `prometheusIngressAuthSecret` isn't configured
+        - `support["prometheus"]["server"]["ingress"]["tls"]` doesn't exist
+    """
+    cluster_config_dir_path = find_absolute_path_to_cluster_file(cluster_name).parent
+
+    config_file = cluster_config_dir_path.joinpath("support.values.yaml")
+    with open(config_file) as f:
+        support_config = yaml.load(f)
+
+    # Don't return the address if the prometheus instance wasn't securely exposed to the outside.
+    if not support_config.get("prometheusIngressAuthSecret", {}).get("enabled", False):
+        raise ValueError(
+            f"`prometheusIngressAuthSecret` wasn't configured for {cluster_name}"
+        )
+
+    tls_config = (
+        support_config.get("prometheus", {})
+        .get("server", {})
+        .get("ingress", {})
+        .get("tls", [])
+    )
+
+    if not tls_config:
+        raise ValueError(
+            f"No tls config was found for the prometheus instance of {cluster_name}"
+        )
+
+    # We only have one tls host right now. Modify this when things change.
+    return tls_config[0]["hosts"][0]
+
+
+def get_cluster_prometheus_creds(cluster_name):
+    """
+    Retrieve the credentials of the prometheus instance running on the `cluster_name` cluster.
+    These credentials are stored in `enc-support.secret.values.yaml` file of each cluster config directory.
+
+    Args:
+        cluster_name: name of the cluster
+    Returns:
+        dict object: {username: `username`, password: `password`}
+    """
+    cluster_config_dir_path = find_absolute_path_to_cluster_file(cluster_name).parent
+
+    config_filename = cluster_config_dir_path.joinpath("enc-support.secret.values.yaml")
+
+    with get_decrypted_file(config_filename) as decrypted_path:
+        with open(decrypted_path) as f:
+            prometheus_config = yaml.load(f)
+
+    return prometheus_config.get("prometheusIngressAuthSecret", {})
+
+
+def get_central_grafana_token(cluster_name):
+    """
+    Get the access token stored in the `enc-grafana-token.secret.yaml`file
+    for the <cluster_name>'s Grafana.
+    This access token should have enough permissions to create datasources.
+
+    Returns:
+        token: the grafana_token stored in the `enc-grafana-token.secret.yaml` file
+    """
+    # Get the location of the file that stores the central grafana token
+    cluster_config_dir_path = find_absolute_path_to_cluster_file(cluster_name).parent
+
+    grafana_token_file = (cluster_config_dir_path).joinpath(
+        "enc-grafana-token.secret.yaml"
+    )
+
+    # Read the secret grafana token file
+    with get_decrypted_file(grafana_token_file) as decrypted_file_path:
+        with open(decrypted_file_path) as f:
+            config = yaml.load(f)
+
+    return config["grafana_token"]
+
+
+def update_central_grafana_token(cluster_name, token):
```

Comment:
Only this function is new. All the others used to exist in `central_grafana.py`.
There are only minor tweaks to the existing ones, like updating the function decription and fixing a small bug https://github.com/2i2c-org/infrastructure/pull/1938/commits/fe182d700f9533cf1974531f81b9e31944cb70c3